docs/DeveloperGuide.md

```diff
@@ -2,32 +2,285 @@
+  custom behaviour.
+* The command first reads a JSON file to obtain the names via `createJsonObject()` method from the
+  superclass.
+* The `trimString` method then removes the `add` command and checks whether the user gave any input after the command. 
+  The method would return the user's input without the command.
+* This input is then passed into the `parseAddCommand()` method to obtain the relevant information: NUS course code, 
+  name of partner university and partner university course code.
+* Along with the JSON Object created from the `createJSONObject()` method, the information extracted from the 
+ `parseAddCommand()` method would be passed to the `isValidInput()` method to verify the user's course mapping.
+* In the `isValidInput()` method, the `getPUCourseList()` method is called to verify the user's partner university is 
+  included in the dataset. An exception is thrown if the university is not found in the dataset.
+* Afterward, the `isValidCourseMapping` checks whether the NUS course code and PU course code are compatible for 
+  course mapping. 
+* If both checks above are passed, the course mapping would be added to the `myList.json` file.
+* Throughout the code, exceptions, assertions and logging are in place for better error handling. 
+* Line Separator is used to ensure readability and ease of use for users.
+
+#### Sequence Diagram on PlantUML
+![Add Courses Sequence Diagram](images/AddCoursesCommand.png)
+
+### 5. Delete Courses Command
+
+#### Overview:
+This command is responsible for deleting users' existing course mapping plan from the `myList.json` file.
+This helps the users to keep track of their most recent course mapping plans, and to keep the `myList.json` file organised.
+
+#### How the feature is implemented:
+* The `DeleteCoursesCommand` class extends `Command` class where it overrides the `execute` method for
+  custom behaviour.
+* When `execute` is called, the command first passes the user's input into the `parseDeleteCommand` method, which parses
+  the user input to extract the list index, still of `String` type, of the course mapping plan 
+  they would like to delete.
+* The parsed input is then passed into the `deleteCourse` method, along with the Storage object, which updates the
+  `myList.json` file. The list index is then converted to an `int` using the `Integer` class. If a valid list index
+  has been given by the user, the list index is passed into the storage object's `deleteCourse` method to delete the 
+  plan stored at that index.
+* Lastly, the UI object's `printDeleteMessage` is called to inform the user of the course plan which is deleted.
+* Throughout the code, exceptions, assertions and logging are in place for better error handling.
+* Line Separator is used to ensure readability and ease of use for users.
+
+#### Sequence Diagram on PlantUML
+![Delete Courses Sequence Diagram](images/DeleteCoursesCommand.png)
```

Comment:
Perhaps it would be better to standardize whether or not there are activation bars?